src/nstime.c

```diff
@@ -141,8 +141,12 @@ nstime_monotonic_t *nstime_monotonic = JEMALLOC_N(n_nstime_monotonic);
 #endif
 bool
 nstime_update(nstime_t *time) {
-	nstime_t old_time;
+	if (nstime_monotonic()) {
+		nstime_get(time);
+		return false;
+	}
 
+	nstime_t old_time;
 	nstime_copy(&old_time, time);
 	nstime_get(time);
 
```

Comment:
Oh I see. Didn't think much about the computed case. There wasn't any issue but I was thinking about calling this during lock spinning, and got worried about the cost (mainly about clock_gettime). I noticed we have the clock_monotonic const so I thought we should either trust the config and avoid a couple instructions here, or have a warning / fatal if we do detect skewed clock. 

Currently the return value can be used to detect such case. Though this may silent the failure as well. I don't think it's a real problem. So I'll just drop this commit if that sounds good to you.